client/src/connection/rest/RestContentAdapter.ts

```diff
@@ -186,21 +202,59 @@ class RestContentAdapter implements ContentAdapter {
     if (!item) {
       return "";
     }
+    const baseKey = item.uri || item.id || item.name;
```

Comment:
This seems like a lot. I'd probably update things such that we're setting the values here, and updating in calling functions where we know the full path might change (rename, move, delete, restore). So the general outline might be something like...
```
// In a function that changes the value
this.pathCache.set(item.id, await this.calculatePathOfItem(item));
```
then the implementation would use the id for the cache key since that's mostly unchanging...
```
if (this.pathCache.has(item.id)) {
      return this.pathCache.get(item.id)!;
    }

    try {
      const path = await this.calculatePathOfItem(item, folderPathOnly);
      this.pathCache.set(item.id, path);
      return path;
    } catch {
      return "";
    }
```

Would something like this work?